src/FsToolkit.ErrorHandling.TaskResult/TaskOptionCE.fs

```diff
@@ -57,63 +56,66 @@ module TaskOptionCE =
 
         member inline _.Delay
             (generator: unit -> Ply<Option<_>>) =
-          uply.Delay(generator)
+          task.Delay(generator)
 
         member inline _.Combine
           (computation1: Ply<Option<'T>>,
            computation2: unit -> Ply<Option<'U>>)
-          : Ply<Option<'U>> = uply {
-            match! computation1 with
-            | None -> return None
-            | Some _ -> return! computation2()
-          }
+          : Ply<Option<'U>> =
+            task {
```

Comment:
revert

Reply:
@NinoFloris I was under the impression the unsafe version ignores the context as well? Or are they just restricted to all safe or unsafe coming from non-affine modules?
https://github.com/crowded/ply#execution-bubble

Reply:
The execution context is not the same as the sync context/scheduler. The unsafe CEs don't capture the previous execution context at the start of the CE (captured to be put back when the CE sets a result on the Task), which can cause AsyncLocals to leak into the outer scope, that's the gist of safe vs unsafe. However when Run uses a safe builder to execute the internal ply returning continuation the execution context will be properly captured and reset by that safe builder.

For the sync context/scheduler all that matters is Affine/NonAffine.

Reply:
Thanks for clearing that up